### ISSUE TITLE
Can I use the drive rather than image as input?

### ISSUE THREAD

**Issue by gilandose** [andlabs/reallymine#8]
I have 2TB drive I can run ddrescue on to an internal 3TB but then I don't have a big enough space to put the output of reallymine... Is it possible to reallymine /dev/sda somefile.img or is this dangerous? If so any other ideas?

**Comment by MrDecay** [comment]
It does work. But if you hit a bad sector it will abort the process.  But
you mileage may vary

**Comment by gilandose** [comment]
Ok thanks, what's typical time to find key sector, I've left running for a few hours and had no output? Is there a quick way to test for finding the key sector? a smaller dump etc?

**Comment by MrDecay** [comment]
When I have used it it usually found it right away and started the
decryption process

Well let me get some info on your harddrive

The typical physical scenario

Your western digital external fails to read to detect.
In preparation you have to remove the encryption bridge.
3.5 hardrives have a removable usb bridge.  2.5 inch drives have the USB
chip directly on the harddrive it self. A little trickier to get to the
data sectors.

If you try to run reallymine on the drive in its normal way. Then there is
no encrypt/decrypt sector available. .. I can continue but lets go one step
at a time 😀

On Sep 15, 2016 6:28 PM, "Richard Gilmore" <notifications@github.com> wrote:

Ok thanks, what's typical time to find key sector, I've left running for a
few hours and had no output? Is there a quick way to test for finding the
key sector? a smaller dump etc?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/andlabs/reallymine/issues/8#issuecomment-247482839>,
or mute
the thread

**Comment by MrDecay** [comment]
https://github.com/andlabs/reallymine/issues/1

Read towards the bottom look for the part of cryptmount might apply to you

**Comment by gilandose** [comment]
Hi sorry I should explain, I already have the harddisk connected to PC via SATA, it is a 3TB disk. The disk stopped being picked up by the computer I'm assuming it is the bridge that has failed.

**Comment by MrDecay** [comment]
Drive spinning and not clicking correct?

On Sep 15, 2016 6:51 PM, "Richard Gilmore" <notifications@github.com> wrote:

Hi sorry I should explain, I already have the harddisk connected to PC via
SATA, it is a 3TB disk. The disk stopped being picked up by the computer
I'm assuming it is the bridge that has failed.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/andlabs/reallymine/issues/8#issuecomment-247486092>,
or mute
the thread

**Comment by gilandose** [comment]
1. Yeah drives working fine as far as I can tell, I'm running ddresue atm and have zero errors with 100GB copied.
2.Had a read through that post that seems like a good way to go about it. I seen the list of possible sectors for the key, is there a linux tool to dump a sector and those round about it. Thanks for the help, sorry I know a good bit more about crypto than I do general hard disk stuff...

**Comment by gilandose** [comment]
`00000000  57 44 76 31 dc c1 00 00  00 88 df e8 00 00 00 00  |WDv1............|
00000010  00 00 00 00 00 00 f0 00  00 00 00 00 00 00 00 00  |................|
00000020  01 00 00 00 00 00 46 50  00 00 00 00 00 00 00 00  |......FP........|
00000030  00 02 ff 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000040  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000050  20 00 0d 7a 00 00 00 01  00 00 00 00 57 44 76 31  | ..z........WDv1|
00000060  63 d4 8f 4d fb 20 d0 14  57 3c 25 5f 7c 8b 37 d9  |c..M. ..W<%_|.7.|
00000070  9c 24 94 e0 3a 5f 00 cc  ca f3 e4 f6 cf 33 07 c9  |.$..:_.......3..|
00000080  09 16 82 e8 5c 7b a9 78  d0 d0 61 f3 43 2f ae bb  |....\{.x..a.C/..|
00000090  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000b0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000c0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000d0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000000f0  f8 38 10 a4 6f be 91 80  82 29 9c 7c f7 fa 92 91  |.8..o....).|....|
00000100  89 e6 27 d1 81 af 8a d6  2e ef 86 5e a0 87 0c 3b  |..'........^...;|
00000110  a3 3f cf 18 c2 eb dd 0e  82 9f 19 0e 68 3e f0 d3  |.?..........h>..|
00000120  22 28 9c 7f 73 1e 8b 89  7f 7a 60 91 23 71 36 ca  |"(..s....z`.#q6.|
00000130  e5 0c af 13 37 a0 dd a1  21 40 c6 93 15 32 fb 82  |....7...!@...2..|
00000140  91 68 74 a4 d3 cd 83 7d  7d 67 7e bb 6d 7d de 86  |.ht....}}g~.m}..|
00000150  e8 a2 c0 36 1d 52 1a ed  bd d4 4e e9 c7 c6 dd b0  |...6.R....N.....|
00000160  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000170  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000180  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000190  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001a0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001b0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001c0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001d0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001e0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
000001f0  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|`

**Comment by gilandose** [comment]
Found the key sector, was looking through the code and the paper, it looks a little different from both of those? there is no magic DEK1
[key-sector.txt](https://github.com/andlabs/reallymine/files/475930/key-sector.txt)

**Comment by MrDecay** [comment]
Mm not sure on tools.  Could it be that your drive is the unsupported
Oxford chip?

**Comment by gilandose** [comment]
it is JMS538S

**Comment by MrDecay** [comment]
Snap a picture of your bridge board

**Comment by gilandose** [comment]
![20160916_025610](https://cloud.githubusercontent.com/assets/7292811/18573241/83d0da0e-7bb9-11e6-8a62-642b31bfe856.jpg)

**Comment by MrDecay** [comment]
Jmicron

**Comment by gilandose** [comment]
Think I have this sorted, thank you for help will update with relevant info in morning. The key was a large number of sectors from end, and it was reading very slowly

**Comment by MrDecay** [comment]
Hmmm I wondering if your suffering from "wd slow" or possibly dirty board
on the harddrive pcb .

Keep us posted.  Maybe you can help out on the encryption info.

Maybe get reallymine to the decryption on the fly status.

**Comment by gilandose** [comment]
Yea, I'd like to I'm currently doing a PhD in side channel, which is encryption breaking through power leakage of chips we'd actually looked at these from that point of view. Will maybe have a try at the cryptmount technique mentioned on the other thread.

**Comment by MrDecay** [comment]
Welcome.   We all started out by using reallymine for our needs and just
answer the best we can To help others. Pietro wrote reallymine.  I just
answer the best i can.

**Comment by MrDecay** [comment]
How did the  decryption go.

**Comment by gilandose** [comment]
Decryption went fine for the first section, but every so often the hard disk stops reading and I get io errors, and I have to completely power it off and it restarts fine, so I'm creating a ddrescue image(average read rate on the hdd is 35MB/s which seems pretty low.) I've written a little python script it utilizes the AES-NI instruction on intel CPU for inplace decryption when I get the image completed. I've tested it on a smaller image and it works fine.

**Comment by andlabs** [comment]
Cool.

I need to add subcommands to find and dump the key block and encryption keys too.

**Comment by MrDecay** [comment]
Hey Richard , how did it go?

**Comment by gilandose** [comment]
Sorry for the slow reply, once I got the full image the inplace decryption worked a treat was able to recovery everything. The disk was really playing up afterwards, tried to reformat and failed, think it is for scrap.

**Comment by MrDecay** [comment]
Yeah the western digital have that degrading issue when the heads. Start
wearing down.  On the upside a new version of reallymine was released with
more debugging features
